DigitalLearningSolutions.Web/Views/LearningPortal/Current/SetCompleteByDate.cshtml

```diff
@@ -1,96 +1,53 @@
-﻿@using DigitalLearningSolutions.Web.ViewModels.LearningPortal.Current
-@using Microsoft.Extensions.Configuration
-@model CurrentLearningItemViewModel
-@inject IConfiguration Configuration
-
-@{
-  ViewData["Title"] = $"Learning Portal - {Model.Name} - Set completion date";
-  ViewData["Application"] = "Learning Portal";
-  ViewData["HeaderPath"] = $"{Configuration["AppRootPath"]}/LearningPortal/Current";
-  ViewData["HeaderPathName"] = "My Current Activities";
-}
-
-@section NavMenuItems {
-  <partial name="Shared/_NavMenuItems" />
-}
+﻿@using DigitalLearningSolutions.Data.Enums
+@using DigitalLearningSolutions.Web.ViewComponents
+@using DigitalLearningSolutions.Web.ViewModels.LearningPortal.Current
+@model EditCompleteByDateViewModel
 
 @{
+  var errorHasOccurred = !ViewData.ModelState.IsValid;
+  ViewData["Title"] = errorHasOccurred ? $"Error: Learning Portal - {Model.Name} - Set completion date" : $"Learning Portal - {Model.Name} - Set completion date";
   var exampleDate = DateTime.Today + TimeSpan.FromDays(7);
-  string prefillDay;
-  string prefillMonth;
-  string prefillYear;
-  if (Model.CompleteByValidationResult?.DateValid == false) {
-    prefillDay = Model.CompleteByValidationResult?.Day.ToString() ?? "";
-    prefillMonth = Model.CompleteByValidationResult?.Month.ToString() ?? "";
-    prefillYear = Model.CompleteByValidationResult?.Year.ToString() ?? "";
-    prefillDay = prefillDay == "0" ? "" : prefillDay;
-    prefillMonth = prefillMonth == "0" ? "" : prefillMonth;
-    prefillYear = prefillYear == "0" ? "" : prefillYear;
+  var standardHintText = $"For example, {exampleDate.Day} {exampleDate.Month} {exampleDate.Year}. Leave the boxes blank to clear the complete by date.";
+  var courseHintText = Model.Type.Equals(LearningItemType.Course) ? "Activities with a complete by date will remain on your 'Current courses' list until completed. Activities with no complete by date will be removed after 6 months of inactivity." : "";
+  var hintTextLines = new List<string> { standardHintText, courseHintText };
+  var action = "";
+  if (Model.Type.Equals(LearningItemType.Course)) {
+    action = "SetCurrentCourseCompleteByDate";
+  } else if (Model.Type.Equals(LearningItemType.SelfAssessment)) {
+    action = "SetSelfAssessmentCompleteByDate";
   } else {
-    prefillDay = Model.CompleteByDate?.Day.ToString() ?? "";
-    prefillMonth = Model.CompleteByDate?.Month.ToString() ?? "";
-    prefillYear = Model.CompleteByDate?.Year.ToString() ?? "";
+    action = "SetCurrentActionPlanResourceCompleteByDate";

```

Comment:
Probably worth nameof for these action names

Reply:
yep, added nameof